test/distributed/fsdp/test_fsdp_state_dict.py

```diff
@@ -440,30 +443,29 @@ def test_wrong_state_dict_config(self):
 
     @skip_if_lt_x_gpu(2)
     def test_state_dict_with_ignored_modules(self):
-        # Initialize an FSDP-wrapped model with an ignored module
-        model = Model(wrap_fsdp=True).cuda()
+        # Initialize an FSDP-wrapped model with an ignored module that includes
+        # both parameters and a buffer
+        model = Model(wrap_fsdp=True, register_buffer=True).cuda()
         ignored_modules = [model.outer]
-        ignored_param_to_param_name = {
+        ignored_tensor_to_tensor_name = {
             model.outer.bias: "outer.bias", model.outer.weight: "outer.weight",
+            model.outer.buffer: "outer.buffer",
         }
         fsdp_model = FSDP(model, ignored_modules=ignored_modules)
         with FSDP.state_dict_type(fsdp_model, StateDictType.FULL_STATE_DICT):
             sd = fsdp_model.state_dict()
-
         with FSDP.summon_full_params(fsdp_model):
             fsdp_params = deepcopy(list(fsdp_model.parameters()))
         # Check that the ignored parameters are not cloned
-
-        for param, param_name in ignored_param_to_param_name.items():
-            self.assertTrue(param_name in sd)
-            self.assertEqual(param.data_ptr(), sd[param_name].data_ptr())
+        for tensor, tensor_name in ignored_tensor_to_tensor_name.items():
+            self.assertTrue(tensor_name in sd)
+            self.assertEqual(tensor.data_ptr(), sd[tensor_name].data_ptr())
         # Check that the state dict can be loaded into a non-wrapped version of
         # the model
-        nonwrapped_model = Model(wrap_fsdp=False).cuda()
+        nonwrapped_model = Model(wrap_fsdp=False, register_buffer=True).cuda()
```

Comment:
Can we also add a check:

Call state_dict() twice, ensure ignored parameters and buffers have the same data_ptr?